client/rest-high-level/src/main/java/org/elasticsearch/client/dataframe/transforms/SourceConfig.java

```diff
@@ -121,4 +122,32 @@ public int hashCode(){
         int hash = Arrays.hashCode(index);
         return 31 * hash + (queryConfig == null ? 0 : queryConfig.hashCode());
     }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private String[] index;
+        private QueryConfig queryConfig;
+
+        public Builder setIndex(String... index) {
```

Comment:
I'm wondering if "setIndices" sounds more natural.
Leaving it up to you though.